javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,385 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+class LinkedList[E]() extends AbstractSequentialList[E]
+    with List[E] with Deque[E] with Cloneable with Serializable {
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  import LinkedList._
+
+  var head: Node[E] = null
+  var last: Node[E] = null
+
+  var _size = 0
+
+  def getFirst(): E =
+    if (_size <= 0) throw new NoSuchElementException()
+    else peekFirst()
+
+  def getLast(): E =
+    if (_size <= 0) throw new NoSuchElementException()
+    else peekLast()
+
+  def removeFirst(): E = {
+    if (isEmpty()) throw new NoSuchElementException()
+    val ret = head.value
+
+    val oldHead = head
+    head = oldHead.next
+
+    if (head ne null) {
+      head.prev = null
+
+      if (head eq last) {
+        head.next = null
+        last = null
+      }
+    }
+
+    _size -= 1
+    ret
+  }
+
+  def removeLast(): E = {
+    if (isEmpty()) throw new NoSuchElementException()
+    if (last eq null) {
+      removeFirst()
+    } else {
+      val oldLast = last
+
+      if (oldLast.prev eq head) {
+        head.next = null
+        last = null
+      } else {
+        last = oldLast.prev
+        last.next = null
+      }
+      last = oldLast.prev
+      last.next = null
+
+      _size -= 1
+      oldLast.value
+    }
+  }
+
+  def addFirst(e: E): Unit = {
+    val oldHead = head
+
+    head = new Node(e, next = oldHead)
+
+    _size += 1
+
+    if (oldHead ne null) {
+      oldHead.prev = head
+
+      if ((last eq null) && (oldHead.next eq null))
+        last = oldHead
```

Comment:
this is not dead code, we are in the case that we are adding to a list with only one element that is the head, so we swap the the oldHead to the last pointer.
This came out correctly by tests that fail in case of eliminating this two lines